app/code/Bulbulatory/Recommendations/Api/Data/RecommendationInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Bulbulatory\Recommendations\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+
+interface RecommendationInterface extends ExtensibleDataInterface

```

Comment:
The idea behind this interface is that the model class (`\Bulbulatory\Recommendations\Model\Recommendation`) should implement it so it have to contain all getters and setters for the model. Check `The DTO` section here: http://vinaikopp.com/2017/02/18/magento2_repositories_interfaces_and_webapi/ and how it implements the interface.

Reply:
@TM18

> The idea behind this interface is that the model class (\Bulbulatory\Recommendations\Model\Recommendation) should implement it so it have to contain all getters and setters for the model. Check The DTO section here: http://vinaikopp.com/2017/02/18/magento2_repositories_interfaces_and_webapi/ and how it implements the interface.

I made a decision use **Bulbulatory\Recommendations\Model\Recommendation** instead **Bulbulatory/Recommendations/Api/Data/RecommendationInterface**